api/interservice/notifications/service/rules.proto

```diff
@@ -0,0 +1,132 @@
+syntax = "proto3";
+package chef.automate.domain.notifications.service;
+
+option go_package = "github.com/chef/automate/api/interservice/notifications/service";
+////
+// Rules management messages and responses
+////
+message Empty { }
+message SlackAlert {
+  string url = 1;
+}
+
+message WebhookAlert {
+  string url = 1;
+}
+
+message ServiceNowAlert {
+  string url = 1;
+  string secret_id = 2;
+  bool critical_controls_only = 3;
+}
+
+message Rule {
+  enum Event {
+    CCRFailure = 0;
+    CCRSuccess = 1;
+    ComplianceFailure = 2;
+    ComplianceSuccess = 3;
+  }
+
+  string id = 1;
+  string name = 2;
+  Event event = 3;
+  oneof action {
+      SlackAlert SlackAlert = 4;
+      WebhookAlert WebhookAlert = 5;
+      ServiceNowAlert ServiceNowAlert = 6;
+  };
+}
+
+message UsernamePassword {
+  string username = 1;
+  string password = 2;
+}
+
+message SecretId {
+  string id = 1;
+}
+
+message URLValidationRequest {
+  string url = 1;
+  oneof credentials {
+    UsernamePassword username_password = 2;
+    SecretId secret_id = 3;
+    Empty none = 4;
+  }
+}
+
+message URLValidationResponse {
+  enum Code {
+    OK = 0;  // we made the request and it succeeded with a 20x response
+    ERROR = 1; // We made the request and it failed for any reason.
+    INVALID_URL = 4; // the url was not valid.
+    NOTIFICATIONS_UNAVAIALBLE = 98; // we could not make the request due to capacity
```

Comment:
```suggestion
    NOTIFICATIONS_UNAVAILABLE = 98; // we could not make the request due to capacity
```

Reply:
I'll add this to the list for cleanup after the cutover.